src/tools/melt/melt.cs

```diff
@@ -252,6 +253,52 @@ private void MeltModule()
         }
 
         /// <summary>
+        /// Extract binary data from tables with a Name and Data column in them.
+        /// </summary>
+        /// <param name="inputPdb">A reference to a <see cref="Pdb"/> as output.  Paths (Data properties) will be modified in this object.</param>
+        /// <param name="package">The installer database to rip from.</param>
+        /// <param name="exportPath">The full path where files will be exported to.</param>
+        /// <param name="tableName">The name of the table to export.</param>
+        private void MeltBinaryTable(ref Pdb inputPdb, InstallPackage package, string exportPath, string tableName)
+        {
+            if(string.IsNullOrEmpty(tableName))
+            {
+                throw new ArgumentNullException(@"tableName");
+            }
+            if(string.IsNullOrEmpty(exportPath))
+            {
+                throw new ArgumentNullException(@"exportPath");
+            }
+            if(null == package)
+            {
+                throw new ArgumentNullException(@"package");
+            }
+            if(null == inputPdb)
+            {
+                throw new ArgumentNullException(@"inputPdb");
+            }
+
+            Table pdbTable = inputPdb.Output.Tables[tableName];
+            if (null == pdbTable)
+            {
+                throw new ArgumentException(string.Format("Table by the name of {0} was not found in the input PDB.", tableName), "tableAndSubDirectoryName");
```

Comment:
This will crash with an exception if a table doesn't exist. It doesn't really matter if a particular table doesn't exist, so I suggest just returning if the table's null. (If it mattered, you could make it a bool method and return false.)